src/transformers/generation/watermarking.py

```diff
@@ -181,43 +189,40 @@ def __call__(
         self,
         input_ids: torch.LongTensor,
         z_threshold: float = 3.0,
-        return_prediction: bool = True,
         return_dict: bool = False,
     ) -> Union[WatermarkDetectorOutput, np.array]:
         """
         Args:
         input_ids (`torch.LongTensor`):
             The watermark generated text. It is advised to remove the prompt, which can affect the detection.
-        z_threshold (`Dict`, *optional*):
-            Changing this threshold will change the sensitivity of the detector.
-        return_prediction (`bool`):
-            Whether to return user-friendly predictions as a boolean array or not.
-        return_dict (`bool`):
-            Whether to return `~generation.WatermarkDetectorOutput` or not. If not it will return `z_score` if
-            `return_prediction=False` else `prediction`,
+        z_threshold (`Dict`, *optional*, defaults to `3.0`):
+            Changing this threshold will change the sensitivity of the detector. Higher z threshold gives less
+            sensitivity and vice versa for lower z threshold.
+        return_dict (`bool`,  *optional*, defaults to `False`):
+            Whether to return `~generation.WatermarkDetectorOutput` or not. If not it will return boolean predictions,
 
         Return:
-            [`~generation.WatermarkDetectorOutput`] or `np.array`: A [`~utils.ModelOutput`] (if `return_dict=True`
-            otherwise a `np.array`.
+            [`~generation.WatermarkDetectorOutput`] or `np.array`: A [`~generation.WatermarkDetectorOutput`]
+            if `return_dict=True` otherwise a `np.array`.
 
         """
 
+        # Let's assume that if one batch start with `bos`, all batched also do
+        if input_ids[0, 0] == self.bos_token_id:
+            input_ids = input_ids[:, 1:]
+
         if input_ids.shape[-1] - self.processor.context_width < 1:
             raise ValueError(
                 f"Must have at least `1` token to score after the first "
                 f"min_prefix_len={self.processor.context_width} tokens required by the seeding scheme."
             )
-        # Let;s assume that if one batch start with `bos`, all batched also do
-        if input_ids[0, 0] == self.bos_token_id:
-            input_ids = input_ids[:, 1:]
 
         num_tokens_scored, green_token_count = self._score_ngrams_in_passage(input_ids)
         z_score = self._compute_z_score(green_token_count, num_tokens_scored)
 
-        if return_dict:
-            p_value = self._compute_pval(z_score)
-            prediction = z_score > z_threshold
-            confidence = 1 - p_value
+        p_value = self._compute_pval(z_score)
+        prediction = z_score > z_threshold
+        confidence = 1 - p_value
 
         if return_dict:
```

Comment:
```suggestion
        if return_dict:
            p_value = self._compute_pval(z_score)
            prediction = z_score > z_threshold
            confidence = 1 - p_value
```
nit: these values are only used with `return_dict`

Reply:
except "prediction", will commit it with other changes :)